internal/mesh/internal/controllers/sidecarproxy/controller_test.go

```diff
@@ -577,13 +577,11 @@ func (suite *controllerTestSuite) TestController() {
 			webComputedDestinations *pbresource.Resource
 		)
 
-		testutil.RunStep(suite.T(), "proxy state template generation", func(t *testing.T) {
-			// Check that proxy state template resource is generated for both the api and web workloads.
-			retry.Run(t, func(r *retry.R) {
-				suite.client.RequireResourceExists(r, apiProxyStateTemplateID)
-				webProxyStateTemplate = suite.client.RequireResourceExists(r, webProxyStateTemplateID)
-				apiProxyStateTemplate = suite.client.RequireResourceExists(r, apiProxyStateTemplateID)
-			})
+		// Check that proxy state template resource is generated for both the api and web workloads.
+		retry.Run(suite.T(), func(r *retry.R) {
+			suite.client.RequireResourceExists(r, apiProxyStateTemplateID)
+			webProxyStateTemplate = suite.client.RequireResourceExists(r, webProxyStateTemplateID)
+			apiProxyStateTemplate = suite.client.RequireResourceExists(r, apiProxyStateTemplateID)
```

Comment:
(Noob question:) Is moving this out of the subtest steps enough to ensure this runs on each retry, or does it need to be in a separate setup func?

Reply:
Yep, anything in the same test scope is guaranteed to run:
```go
func TestTestutilRunStep(t *testing.T) {
	setup() // always runs

	testutil.RunStep(t, "A", func(t *testing.T) { fmt.Println("A") })
	testutil.RunStep(t, "B", func(t *testing.T) { fmt.Println("B") })
	testutil.RunStep(t, "C", func(t *testing.T) { fmt.Println("C") })
	testutil.RunStep(t, "D", func(t *testing.T) { fmt.Println("D") })

        post() // always runs
}
```

So when you run `TestTestutilRunStep/B`, it will skip any subtests not named "B" and effectively run:
```go
func TestTestutilRunStep(t *testing.T) {
	setup() // always runs

	// testutil.RunStep(t, "A", func(t *testing.T) { fmt.Println("A") })
	testutil.RunStep(t, "B", func(t *testing.T) { fmt.Println("B") })
	// testutil.RunStep(t, "C", func(t *testing.T) { fmt.Println("C") })
	// testutil.RunStep(t, "D", func(t *testing.T) { fmt.Println("D") })

        post() // always runs
}
```

Reply:
(even though it's wrapped in a retry, we never step into a new subtest in the retry closure)